cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -713,11 +736,24 @@ public FatalProblem(String message) {
     }
 
     private boolean canRoll(NodeRef nodeRef, long timeout, TimeUnit unit, boolean ignoreSslError, RestartContext restartContext)
-            throws ForceableProblem, InterruptedException {
+            throws ForceableProblem, InterruptedException, UnforceableProblem {
         try {
-            return await(availability(allClient).canRoll(nodeRef.nodeId()), timeout, unit,
-                t -> new ForceableProblem("An error while trying to determine the possibility of updating Kafka pods", t));
-        } catch (ForceableProblem e) {
+            if (nodeRef.broker() && nodeRef.controller()) {
+                boolean canRollController = await(restartContext.quorumCheck.canRollController(nodeRef.nodeId()), timeout, unit,
+                        t -> new UnforceableProblem("An error while trying to determine the possibility of updating Kafka controller pods", t));
```

Comment:
It would be great to have here some more information about the reason. E.g. to know if this is because not enough controllers are in-sync.

Reply:
We throw this if we couldn't describe the quorum for some reason e.g. admin client disconnect. I think the actual reason will be logged and provided to this exception.